xbrowse/datastore/elasticsearch_datastore.py

```diff
@@ -258,42 +266,103 @@ def get_elasticsearch_variants(
+                genotype_filter = value
+                if type(genotype_filter) == int or type(genotype_filter) == basestring:
+                    genotype_filters[sample_id] = [('term', genotype_filter)]
+                elif '$gte' in genotype_filter:
+                    genotype_filter = {k.replace("$", ""): v for k, v in genotype_filter.items()}
+                    genotype_filters[sample_id] = [('range', genotype_filter)]
+                elif "$in" in genotype_filter:
+                    num_alt_values = genotype_filter['$in']
+                    genotype_filters[sample_id] = [('term', num_alt_value) for num_alt_value in num_alt_values]
 
-            s = s.filter(atleast_one_nonref_genotype_filter)
+        sample_ids = [family_individual_ids_to_sample_ids.get(indiv_id) or indiv_id for indiv_id in (indivs_to_consider or [])]
 
+        min_ab = None
+        min_gq = None
         if quality_filter is not None and indivs_to_consider:
-            #'vcf_filter': u'pass', u'min_ab': 17, u'min_gq': 46
             min_ab = quality_filter.get('min_ab')
             if min_ab is not None:
-                min_ab /= 100.0   # convert to fraction
+                min_ab /= 100.0  # convert to fraction
             min_gq = quality_filter.get('min_gq')
             vcf_filter = quality_filter.get('vcf_filter')
-            for indiv_id in indivs_to_consider:
-                sample_id = family_individual_ids_to_sample_ids.get(indiv_id) or indiv_id
+            if vcf_filter is not None:
+                s = s.filter(~Q('exists', field='filters'))
+
+        if is_nested:
+            #  Subquery for child docs with the requested sample IDs
+            sample_id_q = Q('terms', sample_id=sample_ids)
+
+            # Return inner hits with the requested samples
+            if indivs_to_consider:
+                s = s.filter(Q('has_child', type='genotype', query=sample_id_q, inner_hits={'size': len(indivs_to_consider)}))
+            else:
+                s = s.filter(Q('has_child', type='genotype', query=Q(), inner_hits={'size': MAX_INNER_HITS}))
+
+            if genotype_filters:
+                genotype_q = None
+                for sample_id, queries in genotype_filters.items():
+                    q = None
+                    for (op, val) in queries:
+                        if q:
+                            q |= Q(op, num_alt=val)
+                        else:
+                            q = Q(op, num_alt=val)
+                    q &= Q('term', sample_id=sample_id)
+                    if not genotype_q:
+                        genotype_q = q
+                    else:
+                        genotype_q |= q
+                s = s.filter(Q('has_child', type='genotype', query=genotype_q,  min_children=len(genotype_filters)))
+            elif indivs_to_consider:
+                s = s.filter(Q('has_child', type='genotype', query=(Q(Q('range', num_alt={'gte': 1}) & sample_id_q))))
```

Comment:
I think there should be a single has_child query where the genotype_q also includes the min_ab and min_gq filters.. otherwise you could have one set of child-docs that match the num_alt filters, and another set that match the min_ab/min_gq ones.